### DISCUSSION TITLE
FAN1 PWM Frequency

### DISCUSSION THREAD

**Discussion by luchomasco** [Jyers/Marlin#989] [General Discussion]
Hi, I'm trying to use FAN1 slot to control a Laser. I measured a too low PWM frequency of 7Hz and I need at least 1kHz (10kHz optimal). Is there a way to change that frequency? Is it going to interfere with other things? 
My motherboard is v4.2.2.
Thanks for any all the help you can give me.
Greetings, Luciano.

**Comment by gothcha** [comment]
There is a long thread on this topic here, but sounds like it is not a simple matter of changing a setting!

https://github.com/MarlinFirmware/Marlin/issues/20638

**Reply by luchomasco** [reply]
Hi @gothcha thank you very much for your answer. It seems that has been solved by setting the frequency with a G-Code.
https://github.com/HofaTheRipper/Marlin/commit/99ea1e54fb92eca2abfbb19bbe0cfac172dcbc96
https://github.com/HofaTheRipper/Marlin/blob/0dcbf474ae015df8103c1abd63bba9c8b82a7907/FastPWM.txt
Is it possible to add this modifications to Jyers Firmware?

**Reply by gothcha** [reply]
Interesting, changing the timer setups etc are modifications to the base Marlin code not the display UI so they are not strictly part of the Jyers firmware, however I will leave this response to @Jyers.

**Reply by luchomasco** [reply]
If it is possible by G-Code, may be it is possible to do it by the LCD? It would be awesome if there is a menu in which we could switch to Laser mode with XX Hz PWM and disabling Z Steppers.

**Comment by Jonny11bonk** [comment]
I also found out this. My fans were always doing some weird sounds like I could hear it turning on and off. Especially with my step-down converter, a lot of clicking. Would it be better to add there a capacitor to smooth the output?

**Reply by MariaKonopnicka** [reply]
I assume your buck converter is wired to the 24V power rail and you are using a transistor to PWM the 12V fan? If not then your setup is wrong.

**Reply by kted** [reply]
You don't need a transistor. It ground switched PWM, so this is the way to connect it:
![jl6prwL](https://user-images.githubusercontent.com/21242761/123063228-e2d5db80-d415-11eb-8b28-4cc8600165fd.jpg)

**Reply by luchomasco** [reply]
It is not a good practice to use different grounds. The best option is to use an optocoupler. Look at this scheme I made for you to understand.
![IMG_20210623_193410_edit_167209917576046](https://user-images.githubusercontent.com/85717400/123176816-85b04900-d45a-11eb-82cb-ffa69807c0e1.jpg)

**Comment by asychev** [comment]
Tried to built 1.3.4 with FAST_PWM_FAN enabled and FAST_PWM_FAN_FREQUENCY 31400 - no luck, still the same 7.85 Hz.

**Reply by luchomasco** [reply]
Hi, did you make all the modifications in the firmawer? look at these posts, where you will find the information.
https://github.com/HofaTheRipper/Marlin/commit/99ea1e54fb92eca2abfbb19bbe0cfac172dcbc96
https://github.com/HofaTheRipper/Marlin/blob/0dcbf474ae015df8103c1abd63bba9c8b82a7907/FastPWM.txt
I haven't tested it because I don't know how to compile yet.
Tell how was your luck.

**Reply by asychev** [reply]
OK, managed M9891 to work. Unfortunately, with my fans (Sunon 4020, GDSTIME 5015) results are not satisfactory.

**Reply by luchomasco** [reply]
Making all the changes suggested in the post I sent you, is it enough to be able to change the frequency? Did you have any problems with your fans even without changing the frequency? Do your fans have problems with the high frequency or with the modified firmware?

**Reply by asychev** [reply]
Yes. No. Yes. I could confirm that it is possible to change PWM frequency with the method above - checked with oscilloscope on FAN1 port.

**Reply by luchomasco** [reply]
Thanks! I will try to compile a firmware and then I will try my stock fans and let you know if I have the same problem.
